code/game/turfs/space/space.dm

```diff
@@ -13,17 +13,28 @@
 	dynamic_lighting = 0
 	luminosity = 1
 	can_border_transition = 1
+	var/static/list/parallax_appearances
 
 /turf/space/New()
 	if(loc)
 		var/area/A = loc
 		A.area_turfs += src
-	icon_state = "[((x + y) ^ ~(x * y) + z) % 25]"
-	var/image/I = image('icons/turf/space_parallax1.dmi',"[icon_state]")
-	I.plane = SPACE_DUST_PLANE
-	I.alpha = 80
-	I.blend_mode = BLEND_ADD
-	overlays += I
+
+/turf/space/initialize()
+	if(!parallax_appearances)
+		parallax_appearances = list()
+		for(var/i in 0 to 25)
+			var/I = "[i]"
+			var/image/base_image = image(icon = icon, icon_state = I) //only because byond will not create an image if you do not give it some values
+			base_image.appearance = appearance
+			base_image.icon_state = I
+			var/image/parallax_overlay = image('icons/turf/space_parallax1.dmi', I)
+			parallax_overlay.plane = SPACE_DUST_PLANE
+			parallax_overlay.alpha = 80
+			parallax_overlay.blend_mode = BLEND_ADD
+			base_image.overlays += parallax_overlay
+			parallax_appearances[I] = base_image
+	appearance = parallax_appearances[icon_state]
```

Comment:
You need to either store the above as number indexes by adding 1 to all of them, or access this using ``"[((x + y) ^ ~(x * y) + z) % 25]"``

Reply:
yes, on it